pandas/core/groupby/generic.py

```diff
@@ -1058,7 +1058,7 @@ def _cython_agg_blocks(
                     #  reductions; see GH#28949
                     obj = obj.iloc[:, 0]
 
-                s = get_groupby(obj, self.grouper)
+                s = get_groupby(obj, self.grouper, observed=True)
```

Comment:
can you give a comment on why you are passing this here

Reply:
Done